### DISCUSSION TITLE
About TMD

### DISCUSSION THREAD

**Discussion by zdraemrain** [OpenFAST/openfast#1342] [Q&A]
I am using FAST V8. When I joined the TMD file to the ServoDyn file to run the .fst file, I get the following error. I would like to know how to solve this problem. Thanks a lot.
![216b19d9cb733e708192b065ef51513](https://user-images.githubusercontent.com/95263409/203185383-cadd4a50-e127-4c1d-b44c-91f4a66d188d.png)

**Comment by jjonkman** [comment]
Dear @zdraemrain,

I'm not sure why enabling the TMD causes a problem in your model, but the error you are reporting has been eliminated in newer versions of the software.  I suggest upgrading your model to OpenFAST.

Best regards,

**Reply by zdraemrain** [reply]
Dear @jjonkman, 

Thank you very much. According to what you said, the problem has been solved. But I want to know whether other passive control methods or active control methods can be simulated in OpenFAST, not just limited to TMD damping control. I would appreciate it if you could tell me how to do it.

Best regards.

**Reply by jjonkman** [reply]
Dear @zdraemrain,

OpenFAST supports many active and passive control functions, including blade pitch, generator torque, nacelle yaw, shaft brake, cables/tendons, aerodynamic flow control, TMDs, and tuned liquid-column dampers.  See the ServoDyn and Structural Control documentation on readthedocs for more information: https://openfast.readthedocs.io/en/main/source/user/servodyn/index.html and https://openfast.readthedocs.io/en/main/source/user/servodyn-stc/StC_index.html.

Best regards,
